### PR TITLE
[Importer] import constant tuple

### PR DESCRIPTION
**Pull request by qingyunqu**
This PR will fix https://github.com/llvm/torch-mlir/issues/2084. But there is another thing failed when I add e2e test.

### CODE REVIEW

**Comment by ramiro050** [comment]
What e2e test are you using and what is the error?

**Comment by qingyunqu** [comment]
> What e2e test are you using and what is the error?

I will show the case tomorrow.

**Comment by qingyunqu** [comment]
> What e2e test are you using and what is the error?

```python
import torch
import torch_mlir

class TupleModule(torch.nn.Module):
    def __init__(self):
        super().__init__()
        self.value = (1, 1)

    def forward(self, x):
        self.value = (1, 2)
        return torch.amax(x, self.value[0])

model = TupleModule().eval()
module = torch_mlir.compile(model, torch.randn(3, 4), output_type="torch")
print(module.operation.get_asm())
```

Error message:
```
python exception: Failure while executing pass pipeline:
error: unknown: unsupported by backend contract: module initializers
note: unknown: see current operation: "torch.initialize.global_slots"(%3) {slotSymNames = [@value]} : (!torch.tuple<int, int>) -> ()
note: unknown: this is likely due to InlineGlobalSlots being unable to inline a global slot

For Torch-MLIR developers, the error can be reproduced with:
$ torch-mlir-opt -pass-pipeline='builtin.module(torchscript-module-to-torch-backend-pipeline{backend-legal-ops= extra-library=})' /tmp/TupleModule.mlir
Add '-mlir-print-ir-after-all -mlir-disable-threading' to get the IR dump for debugging purpose.
```

**Comment by ramiro050** [comment]
The `InlineGlobalSlots` pass fails because mutation of module parameters is not supported:

https://github.com/llvm/torch-mlir/blob/aa97c8383e96db645b579f58599255f6f41fa1b9/lib/Dialect/Torch/Transforms/InlineGlobalSlots.cpp#L85-L94

However, you should be able to just do

```
class TupleModule(torch.nn.Module):
    def __init__(self):
        super().__init__()

    def forward(self):
        return (1, 2)
```

to test the functionality of importing a tuple constant.

File: python/torch_mlir/dialects/torch/importer/jit_ir/csrc/node_importer.cpp [hunk 1]
```diff
@@ -233,6 +233,13 @@ void NodeImporter::importNode(Node *node, MlirBlock appendToBlock,
                        dummyAnnotator, importOptions);
       mapResults(node, mlirOpResultGetOwner(listValue));
       return; // Early return, since `importIValue` already added op to block.
+    } else if (output->type()->cast<c10::TupleType>()) {
```
Comment (ramiro050) [inline python/torch_mlir/dialects/torch/importer/jit_ir/csrc/node_importer.cpp:236]: This is the same code as the `ListType` case above. We should just change the `if` condition and reuse the logic.

**Review by ramiro050 — CHANGES_REQUESTED** [review]

**Comment by dvlp-r** [comment]
Hi, there is any news about it? @qingyunqu are you still working on it?

**Comment by qingyunqu** [comment]
> Hi, there is any news about it? @qingyunqu are you still working on it?

This PR is ready now.

File: e2e_testing/xfail_sets.py [hunk 2]
```diff
@@ -540,6 +543,7 @@
     "NormScalarOptDimKeepDimModule_basic",
     "NormScalarOptDimModule_basic",
     "NormalizeModule_basic",
+    "ScalarConstantTupleModule_basic",
```
Comment (ramiro050) [inline e2e_testing/xfail_sets.py:546]: The dynamo tests should either be `xfail` or `crashing`. From the error message, it seems like it is not crashing.

**Review by ramiro050 — APPROVED** [review]
LGTM, just a small change request

File: e2e_testing/xfail_sets.py (hunk 2 above)
Comment (qingyunqu) [reply e2e_testing/xfail_sets.py:546]: I think this change is adding pass case on `STABLEHLO_PASS_SET`. It's not about dynamo.

Comment (ramiro050) [reply e2e_testing/xfail_sets.py:545]: Ah, sorry about that. The way github displays it confused me

**Pull request merged**